### DISCUSSION TITLE
Using web app in bot (react js)

### DISCUSSION THREAD

**Discussion by EhsanBeheshtAein** [telegraf/telegraf#1898] [Q&A]
Hello friends
I tried to connect a React project to Backend written with NodeJS
So that I can open the bot web app and transfer some data from the bot to the web app and vice versa
I did not find an example for this
I tried to implement the code with the documentation, but unfortunately it doesn't work!

Please tell me if you have an example in this field or know a way that will help me. 🙏

**Comment by MKRhere** [comment]
There is a comprehensive doc on implementing Mini Apps here:

[Documentation / Mini Apps](https://github.com/feathers-studio/telegraf-docs/tree/master/examples/mini-apps)

> but unfortunately it doesn't work!

When you say, "it doesn't work!", remember we have no idea how to help you.

You must tell us:

* what you tried to do (steps you took, code that causes the issue)?
* what you expected to happen (behaviour of the bot, etc)?
* what actually happened (error messages, unexpected behaviour)?
